react/features/toolbox/functions.js

```diff
@@ -172,13 +174,21 @@ export function getDefaultToolboxButtons(): Object {
         toolbarButtons
             = interfaceConfig.TOOLBAR_BUTTONS.reduce(
                 (acc, buttonName) => {
-                    const button = defaultToolbarButtons[buttonName];
+                    let button = defaultToolbarButtons[buttonName];
+                    const currentButtonHandlers = buttonHandlers[buttonName];
 
                     if (button) {
                         const place = _getToolbarButtonPlace(buttonName);
 
                         button.buttonName = buttonName;
 
+                        if (currentButtonHandlers) {
+                            button = {
+                                ...button,
+                                ...currentButtonHandlers
+                            };
+                        }
+
```

Comment:
I'm seeing a series of fixes for toolbox where through a config object the button is being set up and the fix entails adding more to the config. How do you foresee the interaction between buttons and toolbar once everything is moved into react? What should happen if there's a state change to a button and no more APP.IO.addListener? My concern is also that the buttons are not being composed declaratively to start but they're essentially moving slowly to be declared through a config object that mimics what the component would look like written out anyway.